### PR TITLE
wifi: esp32: remove posix api dependency

### PR DESCRIPTION
**Pull request by sylvioalves**

### CODE REVIEW

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

File: drivers/wifi/esp32/Kconfig.esp32 [hunk 1]
```diff
@@ -2,15 +2,11 @@
 
 menuconfig WIFI_ESP32
 	bool "ESP32 SoC WiFi support"
-	select POSIX_API
```
Comment (mniestroj) [inline drivers/wifi/esp32/Kconfig.esp32:5]: Isn't this possible only after updating `hal_espressif`? If yes, then commits are in wrong order and break bisectability.

Also "HAL library does no need posix calls by default" could be elaborated. Posix were never needed? Or the recent HAL update allows to drop posix calls?

Comment (sylvioalves) [reply drivers/wifi/esp32/Kconfig.esp32:5]: @mniestroj change the commit order. Thanks.
Yes, HAL was update so that POSIX API is not a dependency anymore.

Comment (mniestroj) [reply drivers/wifi/esp32/Kconfig.esp32:5]: Would be good to have "HAL was updated so that POSIX API is not a dependency anymore." in a commit message :)

**Review by mniestroj — APPROVED** [review]

File: drivers/wifi/esp32/Kconfig.esp32 (hunk 1 above)
Comment (sylvioalves) [reply drivers/wifi/esp32/Kconfig.esp32:5]: sure, no problem.

**Pull request merged**